lib/kafka/consumer.rb

```diff
@@ -449,9 +463,9 @@ def seek_to_next(topic, partition)
     end
 
     def resume_paused_partitions!
-      @paused_partitions.each do |topic, partitions|
-        partitions.keys.each do |partition|
-          unless paused?(topic, partition)
+      @pauses.each do |topic, partitions|
+        partitions.each do |partition, pause|
+          if pause.paused? && pause.expired?
```

Comment:
this is a bug. Should be if !pause.paused? || pause.expired?

Reply:
I don't think it is – basically, it's `if partition has been paused but pause has expired`.